src/compas/datastructures/mesh/_mesh.py

```diff
@@ -2165,6 +2165,44 @@ def face_neighbors(self, fkey):
                 nbrs.append(nbr)
         return nbrs
 
+    def face_neighborhood(self, key, ring=1):
+        """Return the faces in the neighborhood of a face.
+
+        Parameters
+        ----------
+        key : hashable
+            The identifier of the face.
+        ring : int, optional
+            The size of the neighborhood.
+            Default is ``1``.
+
+        Returns
+        -------
+        list
+            A list of face identifiers.
+
+        """
+
+        nbrs = set(self.face_neighbors(key))
+
+        i = 1
+        while True:
+            if i == ring:
+                break
+
+            temp = []
+            for key in nbrs:
+                temp += self.face_neighbors(key)
+
+            nbrs.update(temp)
+
+            i += 1
+
+        if key in nbrs:
+            nbrs.remove(key)
+
```

Comment:
is this necessary if you are using a set?

Reply:
nope! I don't know why I added it in the first place!